use rosidl_runtime_c instead of generator

the current master is broken due to the changes introduced in here: https://github.com/ros2/rosidl/pull/456

```
  -- Build files have been written to: /home/runner/work/rmw_iceoryx/rmw_iceoryx/ros_ws/build/rmw_iceoryx_cpp
  Scanning dependencies of target rmw_iceoryx_name_conversion
  [  2%] Building CXX object CMakeFiles/rmw_iceoryx_name_conversion.dir/src/internal/iceoryx_name_conversion.cpp.o
  /home/runner/work/rmw_iceoryx/rmw_iceoryx/ros_ws/src/rmw_iceoryx/rmw_iceoryx_cpp/src/internal/iceoryx_name_conversion.cpp:22:10: fatal error: rosidl_generator_c/primitives_sequence.h: No such file or directory
   #include "rosidl_generator_c/primitives_sequence.h"
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  compilation terminated.
  CMakeFiles/rmw_iceoryx_name_conversion.dir/build.make:62: recipe for target 'CMakeFiles/rmw_iceoryx_name_conversion.dir/src/internal/iceoryx_name_conversion.cpp.o' failed
  make[2]: *** [CMakeFiles/rmw_iceoryx_name_conversion.dir/src/internal/iceoryx_name_conversion.cpp.o] Error 1
  CMakeFiles/Makefile2:131: recipe for target 'CMakeFiles/rmw_iceoryx_name_conversion.dir/all' failed
  make[1]: *** [CMakeFiles/rmw_iceoryx_name_conversion.dir/all] Error 2
  make: *** [all] Error 2
  Makefile:140: recipe for target 'all' failed
  ---
```

--- a/iceoryx_ros2_bridge/src/generic_subscription.cpp
+++ b/iceoryx_ros2_bridge/src/generic_subscription.cpp
@@ -69,7 +69,7 @@ std::shared_ptr<rmw_serialized_message_t> GenericSubscription::create_serialized
 }
 
 void GenericSubscription::handle_message(
-  std::shared_ptr<void> & message, const rmw_message_info_t & message_info)
+  std::shared_ptr<void> & message, const rclcpp::MessageInfo & message_info)
 {
   (void) message_info;
   auto typed_message = std::static_pointer_cast<rmw_serialized_message_t>(message);
@@ -90,7 +90,7 @@ void GenericSubscription::return_serialized_message(
 
 void GenericSubscription::handle_loaned_message(
   void * loaned_message,
-  const rmw_message_info_t & message_info)
+  const rclcpp::MessageInfo & message_info)
 {
   (void) loaned_message;
   (void) message_info;

--- a/iceoryx_ros2_bridge/src/generic_subscription.hpp
+++ b/iceoryx_ros2_bridge/src/generic_subscription.hpp
@@ -72,7 +72,7 @@ public:
   std::shared_ptr<rmw_serialized_message_t> create_serialized_message() override;
 
   void handle_message(
-    std::shared_ptr<void> & message, const rmw_message_info_t & message_info) override;
+    std::shared_ptr<void> & message, const rclcpp::MessageInfo & message_info) override;
 
   // Same as return_serialized_message() as the subscription is to serialized_messages only
   void return_message(std::shared_ptr<void> & message) override;
@@ -80,7 +80,7 @@ public:
   void return_serialized_message(std::shared_ptr<rmw_serialized_message_t> & message) override;
 
   void handle_loaned_message(
-    void * loaned_message, const rmw_message_info_t & message_info) override;
+    void * loaned_message, const rclcpp::MessageInfo & message_info) override;
 
 private:
   RCLCPP_DISABLE_COPY(GenericSubscription)

--- a/iceoryx_ros2_bridge/src/iceoryx_ros2_bridge.cpp
+++ b/iceoryx_ros2_bridge/src/iceoryx_ros2_bridge.cpp
@@ -46,7 +46,7 @@ bool deserialize_into(
   const rosidl_typesupport_introspection_cpp::MessageMembers * introspection_ts,
   void * ros_msg)
 {
-  introspection_ts->init_function(ros_msg, rosidl_generator_cpp::MessageInitialization::ALL);
+  introspection_ts->init_function(ros_msg, rosidl_runtime_cpp::MessageInitialization::ALL);
 
   auto ret = rmw_deserialize(serialized_msg, ts, ros_msg);
   if (ret != RMW_RET_OK) {
@@ -139,7 +139,7 @@ void publish_to_ros2(
         ts, rosidl_typesupport_introspection_cpp::typesupport_identifier)->data);
 
     void * ros_msg = malloc(introspection_ts->size_of_);
-    introspection_ts->init_function(ros_msg, rosidl_generator_cpp::MessageInitialization::ALL);
+    introspection_ts->init_function(ros_msg, rosidl_runtime_cpp::MessageInitialization::ALL);
     rmw_iceoryx_cpp::deserialize(static_cast<const char *>(chunk), introspection_ts, ros_msg);
 
     ret = serialize_into(ros_msg, ts, &serialized_msg);

--- a/iceoryx_ros2_bridge/src/typesupport_helpers.cpp
+++ b/iceoryx_ros2_bridge/src/typesupport_helpers.cpp
@@ -25,7 +25,7 @@
 
 #include "Poco/SharedLibrary.h"
 
-#include "rosidl_generator_cpp/message_type_support_decl.hpp"
+#include "rosidl_runtime_cpp/message_type_support_decl.hpp"
 
 namespace iceoryx_ros2_bridge
 {

--- a/iceoryx_ros2_bridge/src/typesupport_helpers.hpp
+++ b/iceoryx_ros2_bridge/src/typesupport_helpers.hpp
@@ -19,7 +19,7 @@
 #include <tuple>
 #include <utility>
 
-#include "rosidl_generator_cpp/message_type_support_decl.hpp"
+struct rosidl_message_type_support_t;
 
 namespace iceoryx_ros2_bridge
 {

--- a/rmw_iceoryx_cpp/CMakeLists.txt
+++ b/rmw_iceoryx_cpp/CMakeLists.txt
@@ -34,8 +34,8 @@ endif()
 find_package(rcpputils REQUIRED)
 find_package(rcutils REQUIRED)
 find_package(rmw REQUIRED)
-find_package(rosidl_generator_c REQUIRED)
-find_package(rosidl_generator_cpp REQUIRED)
+find_package(rosidl_runtime_c REQUIRED)
+find_package(rosidl_runtime_cpp REQUIRED)
 find_package(rosidl_typesupport_cpp REQUIRED)
 find_package(rosidl_typesupport_introspection_c REQUIRED)
 find_package(rosidl_typesupport_introspection_cpp REQUIRED)
@@ -48,8 +48,8 @@ target_include_directories(rmw_iceoryx_serialization
   PUBLIC include
 )
 ament_target_dependencies(rmw_iceoryx_serialization
-  rosidl_generator_c
-  rosidl_generator_cpp
+  rosidl_runtime_c
+  rosidl_runtime_cpp
   rosidl_typesupport_introspection_c
   rosidl_typesupport_introspection_cpp
 )
@@ -68,8 +68,8 @@ ament_target_dependencies(rmw_iceoryx_name_conversion
   rcpputils
   rcutils
   rmw
-  rosidl_generator_c
-  rosidl_generator_cpp
+  rosidl_runtime_c
+  rosidl_runtime_cpp
   rosidl_typesupport_introspection_c
   rosidl_typesupport_introspection_cpp
 )

--- a/rmw_iceoryx_cpp/package.xml
+++ b/rmw_iceoryx_cpp/package.xml
@@ -14,8 +14,8 @@
   <depend>rcutils</depend>
   <depend>rmw</depend>
   <depend>rmw_implementation</depend>
-  <depend>rosidl_generator_c</depend>
-  <depend>rosidl_generator_cpp</depend>
+  <depend>rosidl_runtime_c</depend>
+  <depend>rosidl_runtime_cpp</depend>
   <depend>rosidl_typesupport_cpp</depend>
   <depend>rosidl_typesupport_introspection_c</depend>
   <depend>rosidl_typesupport_introspection_cpp</depend>

--- a/rmw_iceoryx_cpp/src/internal/iceoryx_deserialize.cpp
+++ b/rmw_iceoryx_cpp/src/internal/iceoryx_deserialize.cpp
@@ -20,7 +20,7 @@
 #include <utility>
 #include <vector>
 
-#include "rosidl_generator_c/primitives_sequence.h"
+#include "rosidl_runtime_c/string_functions.h"
 
 #include "rosidl_typesupport_cpp/message_type_support.hpp"
 
@@ -82,8 +82,8 @@ inline std::pair<const char *, size_t> get_submessage_array_c(
   std::tie(serialized_msg, array_elements) = load_array_size(serialized_msg);
 
   auto data_array =
-    const_cast<rosidl_generator_c__char__Sequence *>(reinterpret_cast<const
-    rosidl_generator_c__char__Sequence
+    const_cast<rosidl_runtime_c__char__Sequence *>(reinterpret_cast<const
+    rosidl_runtime_c__char__Sequence
     *>(ros_message_field));
 
   data_array->data = static_cast<signed char *>(calloc(array_elements, sub_members_size));
@@ -143,8 +143,8 @@ const char * copy_payload_array_c(const char * serialized_msg, void * ros_messag
   uint32_t char_size = size * array_size;
 
   auto data_array =
-    const_cast<rosidl_generator_c__char__Sequence *>(reinterpret_cast<const
-    rosidl_generator_c__char__Sequence
+    const_cast<rosidl_runtime_c__char__Sequence *>(reinterpret_cast<const
+    rosidl_runtime_c__char__Sequence
     *>(ros_message_field));
 
   data_array->data = static_cast<signed char *>(calloc(array_size, size));
@@ -166,8 +166,8 @@ const char * copy_payload_array_c<std::string>(
   serialized_msg = copy_payload_array_c<char>(serialized_msg, ros_message_field);
 
   auto data_array =
-    const_cast<rosidl_generator_c__char__Sequence *>(reinterpret_cast<const
-    rosidl_generator_c__char__Sequence
+    const_cast<rosidl_runtime_c__char__Sequence *>(reinterpret_cast<const
+    rosidl_runtime_c__char__Sequence
     *>(ros_message_field));
 
   --data_array->capacity;

--- a/rmw_iceoryx_cpp/src/internal/iceoryx_name_conversion.cpp
+++ b/rmw_iceoryx_cpp/src/internal/iceoryx_name_conversion.cpp
@@ -19,9 +19,6 @@
 
 #include "rcpputils/split.hpp"
 
-#include "rosidl_generator_c/primitives_sequence.h"
-#include "rosidl_generator_c/primitives_sequence_functions.h"
-
 #include "rosidl_typesupport_cpp/message_type_support.hpp"
 
 #include "rosidl_typesupport_introspection_c/field_types.h"

--- a/rmw_iceoryx_cpp/src/internal/iceoryx_serialize.cpp
+++ b/rmw_iceoryx_cpp/src/internal/iceoryx_serialize.cpp
@@ -17,7 +17,7 @@
 #include <string>
 #include <vector>
 
-#include "rosidl_generator_c/primitives_sequence.h"
+#include "rosidl_runtime_c/string_functions.h"
 
 #include "rosidl_typesupport_cpp/message_type_support.hpp"
 
@@ -66,7 +66,7 @@ size_t get_array_elememts_and_assign_ros_message_field_c(
   void * & subros_message
 )
 {
-  auto data_array = reinterpret_cast<const rosidl_generator_c__char__Sequence *>(ros_message_field);
+  auto data_array = reinterpret_cast<const rosidl_runtime_c__char__Sequence *>(ros_message_field);
   uint32_t array_size = data_array->size;
 
   if (member->is_upper_bound_ && array_size > member->array_size_) {
@@ -107,7 +107,7 @@ void copy_data_array_cpp<bool>(std::vector<char> & payloadVector, const void * r
 template<typename T>
 void copy_data_array_c(std::vector<char> & payloadVector, const void * ros_message_field)
 {
-  auto data_array = reinterpret_cast<const rosidl_generator_c__char__Sequence *>(ros_message_field);
+  auto data_array = reinterpret_cast<const rosidl_runtime_c__char__Sequence *>(ros_message_field);
   uint32_t array_size = data_array->size;
   uint32_t size = sizeof(T) * array_size;
 
@@ -179,7 +179,7 @@ void copy_data_c<std::string>(
       }
     } else {
       auto data_array =
-        reinterpret_cast<const rosidl_generator_c__char__Sequence *>(ros_message_field);
+        reinterpret_cast<const rosidl_runtime_c__char__Sequence *>(ros_message_field);
       uint32_t array_size = data_array->size;  // number of strings in the array
 
       store_array_size(payloadVector, array_size);

--- a/rmw_iceoryx_cpp/src/internal/iceoryx_type_info_introspection.cpp
+++ b/rmw_iceoryx_cpp/src/internal/iceoryx_type_info_introspection.cpp
@@ -17,8 +17,6 @@
 #include <unordered_map>
 #include <utility>
 
-#include "rosidl_generator_c/primitives_sequence.h"
-
 #include "rosidl_typesupport_cpp/message_type_support.hpp"
 
 #include "rosidl_typesupport_introspection_c/field_types.h"
@@ -251,7 +249,7 @@ void iceoryx_init_message(
   if (ts_cpp != nullptr) {
     auto members =
       static_cast<const rosidl_typesupport_introspection_cpp::MessageMembers *>(ts_cpp->data);
-    members->init_function(message, rosidl_generator_cpp::MessageInitialization::ALL);
+    members->init_function(message, rosidl_runtime_cpp::MessageInitialization::ALL);
     return;
   }
 

